kimchi/src/linearization.rs

```diff
@@ -199,6 +199,16 @@ pub fn constraints_expr<F: PrimeField + SquareRootField>(
         assert_eq!(generic_alphas.next(), Some(0));
     }
 
+    // Check that the feature flags correctly turn on or off the constraints generated by the given
+    // flags.
+    if cfg!(feature = "check_feature_flags") {
```

Comment:
this could just be `if cfg!(debug_assertions) {`, unless you think people are going to want to run this in prod